apis/projectcontour/v1alpha1/contourconfig.go

```diff
@@ -75,31 +74,48 @@ type ContourConfigurationSpec struct {
 
 	// Metrics defines the endpoint Contour uses to serve metrics.
 	// +optional
-	// +kubebuilder:default={address: "0.0.0.0", port: 8000}
-	Metrics MetricsConfig `json:"metrics"`
+	// +kubebuilder:default={port: 8002}
+	Metrics *MetricsConfig `json:"metrics"`
 }
 
 // XDSServerType is the type of xDS server implementation.
 type XDSServerType string
```

Comment:
for this sort of thing I also see `+enum` as an annotation (e.g. [here](https://github.com/kubernetes/api/blob/e6d62ddcb18470ddd34d9c4cd0e3178ab58cfaeb/core/v1/types.go#L2610)), which is described [here](https://github.com/kubernetes/enhancements/issues/2887), not sure if we care to do this or not, might make things nice for downstream consumers? (but also not clear to me if this locks you into the same thing as kubebuilder enum, that adding values is a major change)

Reply:
The argument against using Enums is that Clients can (and should) assume that they know all possible values of the enum, so adding or removing fields can break clients unless it's made very clear that the enum could be changed. Making the type an aliased string leaves the door open for other values than the list at any particular version.

It feels to me like this applies no matter how the enum is specified.

Does this really apply here? I think so, but I suppose that a lot of these fields are unlikely to have further values added. I think XDSServerType is a good example of one where we could conceivably add further values (like maybe `incremental` or something) as we add further functionality and want to have a chance to get it in people's hands before it's as stable as the rest.

I agree that having to document this stuff manually sucks, but this feels to me like another time when the rule has been written in upstream maintainer pain.